tublerone/R/dedup_clones.R

```diff
@@ -1,18 +1,39 @@
-dedup_clones = function(snp_pairwise) {
+#' @title dedup_clones
+#'
+#' @description Deduplicates clones which have a minimum distance between them. 
+#'
+#' @param snp_pairwise Two-dimensional named matrix of pairwise-distances.
+#' @param minimum_dist Minimum distance to consider two sequences a clone.
+#'
+#' @return matrix
+#'
+#' @examples
+#' test = rbind(c(0,1,5,6,5),
+#'              c(2,0,5,6,7),
+#'              c(4,5,0,0,0),
+#'              c(4,5,0,0,0),
+#'              c(4,5,0,0,0),
+#' rownames(test) = paste0(rep('seq', 5), seq(1,5))
+#' colnames(test) = paste0(rep('seq', 5), seq(1,5))
+#' dedup_clones(test)
+#'
+#' @export 
 
-identicals = list()
-for (row in rownames(snp_pairwise)) {
-  identicals[[row]] = sort(colnames(snp_pairwise)[which(snp_pairwise[row, ] == 0)])
+dedup_clones = function(snp_pairwise, minimum_dist = 0) {
+
+  identicals = list()
+  for (row in rownames(snp_pairwise)) {
+    identicals[[row]] = sort(colnames(snp_pairwise)[which(snp_pairwise[row, ] == minimum_dist)])
```

Comment:
The logic here is wrong. As written, duplicates will only be removed if the pairwise distance is equal to `minimum_dist`. This is not the intended functionality.